qa/common/util.py

```diff
@@ -0,0 +1,117 @@
+# limitations under the License.
+
+import os
+import subprocess
+
+util_sh_path = os.path.abspath("../common/util.sh")
+
+
+def check_test_results(test_log_path, expected_number_of_test):
+    try:
+        subprocess.run([
+            "bash", "-c", "source " + util_sh_path + "; check_test_results \"" +
+            str(test_log_path) + "\" \"" + str(expected_number_of_test) + "\""
+        ],
+                       check=True)
+    except subprocess.CalledProcessError:
+        return False
+    return True
+    """
+    f = open(test_log_absolute_path, "r+")
+
+    log_contents = f.read()
+
+    total = 0
+    errors = 0
+    failures = 0
+
+    for row in log_contents.split("\n"):
+        # Look for rows like: "{........}"
+        if len(row) > 0 and row[0] == "{" and row[-1] == "}":
+            # Look for key-value pairs
+            key_val_pair = row[1:-1].split(", ")
+            for i in range(0, 3):
+                # Extract key-value pairs from last matching row
+                (key, val) = key_val_pair[i].split(": ")
+                if key == "\"total\"":
+                    total = int(val)
+                elif key == "\"errors\"":
+                    errors = int(val)
+                elif key == "\"failures\"":
+                    failures = int(val)
+
+    if total == expected_number_of_test and errors == 0 and failures == 0:
+        f.close()
+        return True
+    
+    f.write("\n***\n*** Test Failed: Expected " + str(expected_number_of_test) + " test(s), " + str(total) + " test(s) executed, " + str(errors) + " test(s) had error, and " + str(failures) + " test(s) failed. \n***")
+    
+    f.close()
+    return False
+    """
+
+
+def check_csv_table_row_column(csv_file, expected_num_columns,
+                               expected_num_rows, tag):
+    try:
+        subprocess.run([
+            "bash", "-c",
+            "source " + util_sh_path + "; check_csv_table_row_column \"" +
+            str(csv_file) + "\" \"" + str(expected_num_columns) + "\" \"" +
+            str(expected_num_rows) + "\" \"" + str(tag) + "\""
+        ],
+                       check=True)
+    except subprocess.CalledProcessError:
+        return False
+    return True
+
+
+def get_number_of_rows_logfile(log_file):
+    try:
+        subprocess.run([
+            "bash", "-c", "source " + util_sh_path +
+            "; get_number_of_rows_logfile \"" + str(log_file) + "\""
+        ],
+                       check=True)
+    except subprocess.CalledProcessError:
+        return False
+    return True
+
+
+def get_number_of_rows_csv(csv_file):
+    try:
+        subprocess.run([
+            "bash", "-c", "source " + util_sh_path +
+            "; get_number_of_rows_csv \"" + str(csv_file) + "\""
+        ],
+                       check=True)
+    except subprocess.CalledProcessError:
+        return False
+    return True
+
+
+def check_log_table_row_column(log_file, expected_num_columns,
```

Comment:
These are fine for now, but there must be more readable pythonic ways to implement each of these?

Reply:
Will not rely on Python script, and complete solely using Bash script. This file will be removed.